File: XDrawer/OnePointFigure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing.Drawing2D;

namespace XDrawer
{
    public abstract class OnePointFigure : Figure
    {
        protected static int DELTA = 5;
        protected int _x1;
        protected int _y1;
        public OnePointFigure(Popup popup, int x, int y)
            : base(popup)
        {
            _x1 = x;
            _y1 = y;
        }
        public override void setXY2(int newX, int newY)
        {
            _x1 = newX;
            _y1 = newY;
        }
        public override void makeRegion()
        {
            System.Drawing.Point[] pt = new System.Drawing.Point[4];
            pt[0].X = _x1 - DELTA; pt[0].Y = _y1 - DELTA;
            pt[1].X = _x1 + DELTA; pt[1].Y = _y1 - DELTA;
            pt[2].X = _x1 + DELTA; pt[2].Y = _y1 + DELTA;
            pt[3].X = _x1 - DELTA; pt[3].Y = _y1 + DELTA;

            byte[] type = new byte[4];
            type[0] = (byte)PathPointType.Line;
            type[1] = (byte)PathPointType.Line;
            type[2] = (byte)PathPointType.Line;
            type[3] = (byte)PathPointType.Line;

            GraphicsPath gp = new GraphicsPath(pt, type);
            _region = new Region(gp);
        }
    }
}

File: XDrawer/OnePointFigure.cpp
#include "stdafx.h"
#include "OnePointFigure.h"

IMPLEMENT_SERIAL(OnePointFigure, Figure, 1)
OnePointFigure::OnePointFigure()
	: Figure()
{
	m_x1 = m_y1 = 0;
}
OnePointFigure::OnePointFigure(int x1, int y1)
	: Figure()
{
	m_x1 = x1;
	m_y1 = y1;
}
void OnePointFigure::Serialize(CArchive& ar)
{
	Figure::Serialize(ar);
	if (ar.IsStoring()) {
		ar << m_x1 << m_y1;
	} else {
		ar >> m_x1 >> m_y1;
	}
}
void OnePointFigure::setXY2(int x, int y)
{
	m_x1 = x;
	m_y1 = y;
}
OnePointFigure::~OnePointFigure()
{
}

void OnePointFigure::makeRegion()
{
	if(region != NULL) delete region;

	region = new CRgn();
	region->CreateRectRgn(m_x1-DELTA, m_y1-DELTA, m_x1+DELTA, m_y1+DELTA);
}
